test/e2e/operator/deployment_api.go

```diff
@@ -352,7 +486,137 @@ func startOperator(c *deploy.Cluster, d *deploy.Deployment) {
 		return false
 	}, "3m", "1s").Should(BeTrue(), "increase operator deployment replicas to 1")
 
-	framework.Logf("Ensure operator pod is eady.")
+	framework.Logf("Ensure operator pod is ready.")
 	deploy.WaitForOperator(c, d.Namespace)
 	framework.Logf("Operator is restored!")
 }
+
+func switchDeploymentMode(c *deploy.Cluster, f *framework.Framework, depName string, mode api.DeviceMode) api.Deployment {
+	By(fmt.Sprintf("Switching driver mode to '%s'", mode))
+	deployment := deploy.GetDeploymentCR(f, depName)
+	deployment.Spec.DeviceMode = mode
+	deployment = deploy.UpdateDeploymentCR(f, deployment)
+
+	waitForDriverReady(c, f, depName)
+
+	return deployment
+}
+
+// waitForDriverReady checks if all node driver pods are in Ready state
```

Comment:
Can you use `WaitForPMEMDriver` instead (https://github.com/intel/pmem-csi/blob/7e2f199db11376f44efcdbb7245b4b9a7f81353e/test/e2e/deploy/deploy.go#L80)?

This looks like a less complete version of that function.

If `WaitForPMEMDriver` needs some additional parameters to be suitable, then just add them.

Reply:
> If `WaitForPMEMDriver` needs some additional parameters to be suitable, then just add them.

Does the existing function(in general metrics logic) work even for the deployment update that resulted in the restart of the only deployment set?

Reply:
It doesn't wait for a restart (that's what you need to deal with here), but once you have verified that the old controller pod is gone, `WaitForPMEMDriver` should work.

Reply:
Pushed a commit (https://github.com/intel/pmem-csi/pull/637/commits/114f71ef2fadbb4cf3757eccf58da65fe8ea50bd) to unregister a node on exit. So that, metrics from controller get updated on the daemon set controller restarts the node pods. This should make `WaitForPMEMDriver` work in case of driver updates only resulted in restart node pods.